x-pack/plugins/security/server/authentication/authentication_result.ts

```diff
@@ -113,17 +113,29 @@ export class AuthenticationResult {
   /**
    * Produces `AuthenticationResult` for the case when authentication needs user to be redirected.
    * @param redirectURL URL that should be used to redirect user to complete authentication.
+   * @param [user] Optional user information retrieved as a result of successful authentication attempt.
```

Comment:
__note:__ we didn't need these before in `redirect`, since we didn't need user information when we did a redirect. But now we need to "register" the fact of login and create a new session as soon as we're able to authenticate user or, in the future, immediately log user out if they have too many active sessions. Another user case when we may need this is when we use Kerberos - we need to respond authentication response headers _and_ potentially redirect user to Overwritten Session or Access Agreement UI.